subprojects/core/src/integTest/groovy/org/gradle/api/tasks/ExecIntegrationTest.groovy

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks
+
+import org.gradle.integtests.fixtures.AbstractConsoleFunctionalSpec
+import spock.lang.Issue
+
+class ExecIntegrationTest extends AbstractConsoleFunctionalSpec {
+    @Issue("https://github.com/gradle/gradle/issues/2009")
+    def "Exec task output is grouped with it's task output"() {
+        given:
+        buildFile << """apply plugin: 'java'
+
+task run(type: Exec) {
+    dependsOn 'compileJava'
+    commandLine 'java', '-cp', "\$sourceSets.main.runtimeClasspath.asPath", 'Main'
+}
+"""
+
+        and:
+        file("src/main/java/Main.java") << """public class Main {
+    public static void main(String[] args) {
+        System.out.println("Hello, World!");
+    }
+}"""
+
+        when:
+        succeeds("run")
+
+        then:
+        (result.output =~ /(?ms)(> Task :run.*?Hello, World!)/).find()
```

Comment:
I think we should cover the different ways to use `Exec`-like things and I'm OK if there are other tests that cover the parallel task execution cases somewhere else (IOW, a different PR someone else is working on or if the coverage already exists).